app/src/main/java/uk/gov/android/securestore/crypto/RsaCryptoManager.kt

```diff
@@ -25,17 +38,38 @@ internal class RsaCryptoManager : CryptoManager {
         return Base64.encodeToString(encryptedBytes, Base64.NO_WRAP)
     }
 
-    override fun decryptText(alias: String, text: String): String {
+    override fun decryptText(
+        text: String,
+        callback: (result: String?) -> Unit,
+        authPromptConfig: AuthenticatorPromptConfiguration?
+    ) {
         val encryptedBytes = Base64.decode(text, Base64.NO_WRAP)
 
-        return Cipher.getInstance(TRANSFORMATION).apply {
-            init(Cipher.DECRYPT_MODE, getKeyEntry(alias).privateKey)
+        val cipher = Cipher.getInstance(TRANSFORMATION)
+
+        if (accessControlLevel == AccessControlLevel.OPEN) {
+            cipher.init(Cipher.DECRYPT_MODE, getKeyEntry(alias).privateKey)
```

Comment:
Lines 51&52 are doing the same thing as lines 60-62. Would it make sense to create a function like
`private fun decryptAndCallback(alias: String, decryptedBytes: ByteArray?, callback: (result: String?) -> Unit`